libalpm/scripts/mkinitcpio

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+ set -e
+
+args=()
+package=0
+
+process_preset() {
+    if [[ -n "${pkgbase}" && -e $preset ]]; then
+        if ! cmp "$preset" > /dev/null 2>&1 <(sed "s|%PKGBASE%|${pkgbase}|g" /usr/share/mkinitcpio/hook.preset); then
+            if [[ ! -e $preset.pacsave ]]; then
+                # save the preset as pacsave
+                mv "$preset" "$preset.pacsave" && return 0
+            fi
+        else
+            # remove the preset
+            rm "$preset" && return 0
+        fi
+    fi
+}
+
+install_kernel() {
+    preset="/etc/mkinitcpio.d/${1}.preset"
+    if [[ ! -e $preset ]]; then
+        if [[ -e $preset.pacsave ]]; then
+            # move the pacsave to the template
+            mv "${preset}.pacsave" "$preset"
```

Comment:
```suggestion
            mv -- "${preset}.pacsave" "$preset"
```

Reply:
This never occurred to me. Is should never be an issue in this context, but still a good idea.